app/src/store/playground/playgroundStore.tsx

```diff
@@ -89,15 +89,13 @@ export const createPlaygroundStore = (
     operationType: "chat",
     inputMode: "manual",
     input: {
-      // TODO(apowell): When implementing variable forms, we should maintain a separate
-      // map of variableName to variableValue. This will allow us to "cache" variable values
-      // as the user types and will prevent data loss if users accidentally change the variable name
-      variables: {
-        // TODO(apowell): This is hardcoded based on the default chat template
-        // Instead we should calculate this based on the template on store creation
-        // Not a huge deal since this will be overridden on the first keystroke
+      // select variablesValueCache and variableKeys from the input in order
```

Comment:
what does this select mean? in what context are we selecting?

Reply:
Good question. I should improve that comment. I meant that in order to see "variables with values" in the playground you need to select those two keys using a zustand selector and combine them. Will rewrite